admin/app/controllers/admin/CommercialController.scala

```diff
@@ -92,6 +96,31 @@ class CommercialController(implicit context: ApplicationContext) extends Control
     }
   }
 
+  def getLineItemsForOrder(orderId: String) = Action { implicit request =>
+    val lineItems: Seq[GuLineItem] = Store.getDfpLineItemsReport().lineItems filter (_.orderId.toString == orderId)
+
+    Cached(5.minutes){
+      JsonComponent(Json.toJson(lineItems))
+    }
+  }
+
+  def getCreativesListing(lineitemId: String, section: String) = Action { implicit request: RequestHeader =>
+
+    val validSections: List[String] = List("uk", "lifeandstyle", "sport", "science")
+
+    val previewUrls: Seq[String] =
+      (for {
+        lineItemId <- Try(lineitemId.toLong).toOption
```

Comment:
It can throw a `NumberFormatException` and since `lineitemId` is supplied as part of the route, this is the only/most sensible place to do the conversion/validation.